### PR TITLE
etcd/cluster-etcd-operator: add steps to migrate quorum-guard

### PR DESCRIPTION
**Pull request by hexfusion**
Below I outline the steps to move quorum-guard to the `openshift-etcd` namespace.

cc @deads2k @alaypatel07 @retroflexer

### CODE REVIEW

**Comment by deads2k** [comment]
why not 
 1. create the the PDB we want in the etcd operator 
 2. actively delete the MCO's PDB. The CVO is very slow at putting it back
 3. then merge a PR to remove the MCO's PDB.

**Comment by hexfusion** [comment]
> why not
> 
> 1. create the the PDB we want in the etcd operator
> 2. actively delete the MCO's PDB. The CVO is very slow at putting it back
> 3. then merge a PR to remove the MCO's PDB.

I thought it might be a stretch adding RBACS for CEO to delete PDB/Deployments in MCO namespace. If that is not an issue then lgtm!

File: enhancements/etcd/cluster-etcd-operator.md [hunk 1]
```diff
@@ -490,7 +490,23 @@ items:
...
+quorum-guard was added to MCO's CVO manifests because CEO did not yet exist. Because we gracefully handle the certs in
+secrets in the `openshift-etcd` namespace it makes sense to move quorum-guard with the addition of the operator.
+
+1.) add quorum-guard deployment to CEO manifests and deploy into `openshift-etcd` namespace mounting the proper secrets.
+2.) remove quorum-guard from CVO controlled manifests in MCO repo.
+3.) create a CVO controlled Jobs that removes the Deployment and PDB from MCO namespace.
+
+#### Downgrade from 4.4 to 4.3
+
+quorum-guard: if another PDB and quorum-guard are running this should not cause an issue. We could document the steps to
```
Comment (deads2k) [inline enhancements/etcd/cluster-etcd-operator.md]: failure of imagination.  If you have another PDB, the MCO will be unable to re-upgrade the machines. You must document these steps.

File: enhancements/etcd/cluster-etcd-operator.md [hunk 2]
```diff
@@ -490,7 +490,23 @@ items:
 
 ### Upgrade / Downgrade Strategy
 
-TODO
+#### Upgrade from 4.3 to 4.4
+
+quorum-guard: currently quorum-guard is a deployment in the `machine-config-operations` namespace the bash logic used
```
Comment (deads2k) [inline enhancements/etcd/cluster-etcd-operator.md]: Add a section above in the main design details that explains what this does, why we have it, and how it works.

Comment (deads2k) [reply enhancements/etcd/cluster-etcd-operator.md]: > Add a section above in the main design details that explains what this does, why we have it, and how it works.

That is valid for 4.4 I mean.

File: enhancements/etcd/cluster-etcd-operator.md [hunk 3]
```diff
@@ -490,7 +490,23 @@ items:
...
+to validate quorum expects certs to exist in /etc/kubernetes/static-pod-resources/etcd-member/. The target controller
+in CEO now installs these certs by revision on disk in a path such as /etc/kubernetes/static-pod-resources/etcd-pod-N/.
+
+quorum-guard was added to MCO's CVO manifests because CEO did not yet exist. Because we gracefully handle the certs in
+secrets in the `openshift-etcd` namespace it makes sense to move quorum-guard with the addition of the operator.
+
+1.) add quorum-guard deployment to CEO manifests and deploy into `openshift-etcd` namespace mounting the proper secrets.
+2.) remove quorum-guard from CVO controlled manifests in MCO repo.
+3.) create a CVO controlled Jobs that removes the Deployment and PDB from MCO namespace.
```
Comment (deads2k) [inline enhancements/etcd/cluster-etcd-operator.md]: blech.  Just do it in the cluster-etcd-operator

File: enhancements/etcd/cluster-etcd-operator.md [hunk 4]
```diff
@@ -490,7 +490,23 @@ items:
...
+
+quorum-guard: currently quorum-guard is a deployment in the `machine-config-operations` namespace the bash logic used
+to validate quorum expects certs to exist in /etc/kubernetes/static-pod-resources/etcd-member/. The target controller
+in CEO now installs these certs by revision on disk in a path such as /etc/kubernetes/static-pod-resources/etcd-pod-N/.
+
+quorum-guard was added to MCO's CVO manifests because CEO did not yet exist. Because we gracefully handle the certs in
+secrets in the `openshift-etcd` namespace it makes sense to move quorum-guard with the addition of the operator.
+
+1.) add quorum-guard deployment to CEO manifests and deploy into `openshift-etcd` namespace mounting the proper secrets.
+2.) remove quorum-guard from CVO controlled manifests in MCO repo.
```
Comment (deads2k) [inline enhancements/etcd/cluster-etcd-operator.md]: this is your cheap intermediate state.  I expect to see it properly managed before 4.5 ships.

File: enhancements/etcd/cluster-etcd-operator.md (hunk 3 above)
Comment (hexfusion) [reply enhancements/etcd/cluster-etcd-operator.md]: we ended up doing a controller to manage to PDB

**Pull request closed without merging**